ast/file_info.go

```diff
@@ -598,6 +597,46 @@ func (pos SourcePos) String() string {
 	return fmt.Sprintf("%s:%d:%d", pos.Filename, pos.Line, pos.Col)
 }
 
+func (pos SourcePos) AsSpan() SourceSpan {
+	return emptySpan{Pos: pos}
+}
+
+type emptySpan struct {
+	Pos SourcePos
+}
+
+func (e emptySpan) Start() SourcePos {
+	return e.Pos
+}
+
+func (e emptySpan) End() SourcePos {
+	return e.Pos
+}
+
+type posSpan struct {
```

Comment:
Since this is really the simplest implementation of `SourceSpan`, I think `sourceSpan` would be more intuitive of a name.